### PR TITLE
Add WatchNotice subclasses and enable streaming refresh

### PR DESCRIPTION
**Pull request by agrare**
Depends on:
- [x] https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/423
- [x] https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/424

### CODE REVIEW

**Review by chessbyte — APPROVED** [review]

**Comment by agrare** [comment]
I'm seeing a lot of endpoint notices, need to see if that is something we need to disable.

**Comment by agrare** [comment]
Yeah so the endpoints are updating every ~2 seconds with the only change being the renewTime:
```
#<Kubeclient::Resource type="MODIFIED", object={:kind=>"Endpoints", :apiVersion=>"v1", :metadata=>{:name=>"kube-controller-manager", :namespace=>"kube-system", :selfLink=>"/api/v1/namespaces/kube-system/endpoints/kube-controller-manager", :uid=>"fab33aa3-e09b-4574-81f3-d1883b9190e9", :resourceVersion=>"1319691", :creationTimestamp=>"2021-01-29T21:12:39Z", :annotations=>{:"control-plane.alpha.kubernetes.io/leader"=>"{\"holderIdentity\":\"ip-172-16-106-133.ec2.internal_220028d7-2717-4a54-84ac-601ffe1a1b6e\",\"leaseDurationSeconds\":15, \"acquireTime\":\"2021-01-29T21:12:39Z\",\"renewTime\":\"2021-02-04T19:09:49Z\",\"leaderTransitions\":0}"}, :managedFields=>[{:manager=>"kube-controller-manager", :operation=>"Update", :apiVersion=>"v1", :time=>"2021-02-04T19:09:49Z", :fieldsType=>"FieldsV1", :fieldsV1=>{:"f:metadata"=>{:"f:annotations"=>{:"."=>{}, :"f:control-plane.alpha.kubernetes.io/leader"=>{}}}}}]}}>
#<Kubeclient::Resource type="MODIFIED", object={:kind=>"Endpoints", :apiVersion=>"v1", :metadata=>{:name=>"kube-controller-manager", :namespace=>"kube-system", :selfLink=>"/api/v1/namespaces/kube-system/endpoints/kube-controller-manager", :uid=>"fab33aa3-e09b-4574-81f3-d1883b9190e9", :resourceVersion=>"1319696", :creationTimestamp=>"2021-01-29T21:12:39Z", :annotations=>{:"control-plane.alpha.kubernetes.io/leader"=>"{\"holderIdentity\":\"ip-172-16-106-133.ec2.internal_220028d7-2717-4a54-84ac-601ffe1a1b6e\",\"leaseDurationSeconds\":15, \"acquireTime\":\"2021-01-29T21:12:39Z\",\"renewTime\":\"2021-02-04T19:09:51Z\",\"leaderTransitions\":0}"}, :managedFields=>[{:manager=>"kube-controller-manager", :operation=>"Update", :apiVersion=>"v1", :time=>"2021-02-04T19:09:51Z", :fieldsType=>"FieldsV1", :fieldsV1=>{:"f:metadata"=>{:"f:annotations"=>{:"."=>{}, :"f:control-plane.alpha.kubernetes.io/leader"=>{}}}}}]}}>
```

This is a crazy amount of churn so I might exclude the endpoints from

**Comment by chessbyte** [comment]
the `resourceVersion` is also changing.  Maybe you can catch this type of update and throw it away?

**Comment by agrare** [comment]
Yeah the resourceVersion is incremented every time there's a change to the resource so that's a given.  If we did want to drop changes to the annotations we'd have to cache the previous to have something to compare to.

File: app/models/manageiq/providers/amazon/container_manager/refresh_worker/runner.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
 class ManageIQ::Providers::Amazon::ContainerManager::RefreshWorker::Runner < ManageIQ::Providers::Kubernetes::ContainerManager::RefreshWorker::Runner
+  private
+
+  def filter_notices!(notices)
```
Comment (Fryguy) [inline app/models/manageiq/providers/amazon/container_manager/refresh_worker/runner.rb:4]: Is the caller for this already implemented in the base class?

Comment (agrare) [reply app/models/manageiq/providers/amazon/container_manager/refresh_worker/runner.rb:4]: Nope this has a dependent PR

Comment (Fryguy) [reply app/models/manageiq/providers/amazon/container_manager/refresh_worker/runner.rb:4]: I see a caller for `filter_notices` (no-bang), but not `filter_notices!`

Comment (agrare) [reply app/models/manageiq/providers/amazon/container_manager/refresh_worker/runner.rb:4]: Yeah I'm thinking of pulling the `Collector#filter_notices` from the base class into the runner, the reason I didn't do this in the collector is by that point we've already started doing a refresh which negates the purpose of reducing churn

**Review by Fryguy — APPROVED** [review]

**Pull request merged**